cmd/fts/cmd/run.go

```diff
@@ -0,0 +1,297 @@
+	runCmd.Flags().String("deals", "deals.csv", "Output file for deals results")
+	runCmd.Flags().String("errors", "errors.csv", "Output file for errors results")
+	runCmd.Flags().Bool("pipe", false, "Pipe all results to stdout")
+	runCmd.Flags().Bool("debug", false, "Debug")
+
+	rootCmd.AddCommand(runCmd)
+}
+
+var runCmd = &cobra.Command{
+	Use:   "run",
+	Short: "Run a storage task pipeline",
+	Long:  `Run a storage task pipeline to migrate large collections of data to Filecoin`,
+	PreRun: func(cmd *cobra.Command, args []string) {
+		err := viper.BindPFlags(cmd.Flags())
+		checkErr(err)
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		maxStagedBytes := viper.GetInt64("maxStagedBytes")
+		maxDealBytes := viper.GetInt64("maxDealBytes")
+		if maxStagedBytes < maxDealBytes {
+			Fatal(fmt.Errorf("Max deal size (%d) is larger than max staging size (%d)", maxDealBytes, maxStagedBytes))
+		}
+
+		// TODO: Implement "file" task input with a list of locations/urls
+		folder := viper.GetString("folder")
+
+		// TODO: Add file size filter
+		alltsks, err := PathToTasks(folder)
+		checkErr(err)
+
+		Message("Found %d tasks", len(alltsks))
+
+		tasks := alltsks[:0]
+
+		hidden := viper.GetBool("hidden")
+		minDealBytes := viper.GetInt64("minDealBytes")
+
+		for _, tsk := range alltsks {
+			if !hidden && strings.HasPrefix(tsk.name, ".") {
+				Message("removing hidden %s", tsk.name)
+				continue
+			}
+			if tsk.bytes < minDealBytes {
+				Message("data too small %s (%d)", tsk.name, tsk.bytes)
+				continue
+			}
+			if maxDealBytes < tsk.bytes {
+				Message("data too large %s (%d)", tsk.name, tsk.bytes)
+				continue
+			}
+			tasks = append(tasks, tsk)
+		}
+
+		Message("Removed %d tasks", len(alltsks)-len(tasks))
+
+		concurrent := viper.GetInt64("concurrent")
+		if int64(len(tasks)) < concurrent {
+			concurrent = int64(len(tasks))
+		}
+
+		rc := RunConfig{
+			token:           viper.GetString("token"),
+			ipfsrevproxy:    viper.GetString("ipfsrevproxy"),
+			serverAddress:   viper.GetString("serverAddress"),
+			maxStagingBytes: maxStagedBytes,
+			minDealBytes:    minDealBytes,
+			concurrent:      concurrent,
+			dryRun:          viper.GetBool("dry-run"),
+		}
+
+		res := Run(tasks, rc)
+
+		if viper.GetBool("pipe") {
+			jobs, deals, errs := resultsToStdOut(res)
+			Success("Complete: %d tasks. %d errors %d deals\n", jobs+errs, errs, deals)
+		} else {
+			jobsFile := viper.GetString("jobs")
+			dealsFile := viper.GetString("deals")
+			errorsFile := viper.GetString("errors")
+			jobs, deals, errs := resultsToCSV(res, jobsFile, dealsFile, errorsFile)
+			Success("Complete: %d tasks. %d errors %d deals\n", jobs+errs, errs, deals)
+		}
+
+	},
+}
+
+func resultsToStdOut(rc chan TaskResult) (int, int, int) {
+	jobs := 0
+	deals := 0
+	errs := 0
+	for {
+		select {
+		case res, ok := <-rc:
+			if ok {
+				if res.err != nil {
+					Message("Error: %s %s %s", res.task.path, res.stage, res.err.Error())
```

Comment:
Nit: `%s` automatically calls `.Error()` if available, so you can use `res.err`.